be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala

```diff
@@ -11,10 +11,10 @@ final case class Channel(channel: String) {
    *
    * @return an array of bytes corresponding to the decoded sub-channel name or None if an error occurred
    */
-  def decodeSubChannel: Option[Array[Byte]] = channel match {
+  def decodeSubChannel: Option[Base64Data] = channel match {
     case _ if channel.startsWith(Channel.ROOT_CHANNEL_PREFIX) =>
       Try(Base64.getUrlDecoder.decode(channel.substring(Channel.ROOT_CHANNEL_PREFIX.length).getBytes)) match {
-        case Success(value) => Some(value)
+        case Success(value) => Some(Base64Data.encode(value))
```

Comment:
are you sure this is not just the following ? 
```suggestion
        case Success(value) => Some(Base64Data(value))
```

Also, this function only works for a channel in the format `/root/myLaoId`, right ?
If I understand correctly, `/root/myLaoId/social` will most likely fail.
How is then not the same as `decodeChannelLaoId` ?

Reply:
indeed, however, this function could be used for subchannel extraction if necessary, but this is why I added the decodeChannelLaoId function.

Reply:
The subchannel extraction will indeed fail, as I looked into it in detail. This function is therefore not useful and will be removed.